### PR TITLE
External logging configured in CM

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Refactoring
- Documentation

### Description
Move External Logging configuration to `valueFrom`.

Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/4247

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by sknot-rh** [comment]
@strimzi-ci run tests profile=regression testcase=LoggingChangeST,RollingUpdateST

**Comment by sknot-rh** [comment]
@strimzi-ci run tests profile=regression testcase=LoggingChangeST,RollingUpdateST

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +14,7 @@
   * `operator.strimzi.io/generation` (used internally only - replaced by `strimzi.io/generation`)
   * `operator.strimzi.io/delete-pod-and-pvc` (use `strimzi.io/delete-pod-and-pvc` instead)
   * `operator.strimzi.io/manual-rolling-update` (use `strimzi.io/manual-rolling-update` instead)
+* External logging configuration has changed. `spec.logging.name` is deprecated. Moved to `spec.logging.valueFrom.configMapKeyRef.name`. Key in the `ConfigMap` is configurable via `spec.logging.valueFrom.configMapKeyRef.key`.
```
Comment (scholzj) [inline CHANGELOG.md:17]: Maybe you can add a simple example of deprecated and new - I think that would help to understand it better. That would serve also as a conversion guide in a way.

File: documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc [hunk 2]
```diff
@@ -60,7 +60,10 @@ spec:
       reconciliationIntervalSeconds: 60
       logging:
         type: external
-        name: customConfigMap
+        valueFrom:
+          configMapKeyRef:
+            name: customConfigMap
+            key: key
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc:66]: Can you use in all these examples a better key example? E.g. `entity-operator-log4j.properties` etc.?

File: documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc [hunk 3]
```diff
@@ -12,7 +12,7 @@ The Topic Operator uses the Apache `log4j2` logger implementation.
 Use the `logging` property in the `entityOperator.topicOperator` field of the Kafka resource `Kafka` resource to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`.
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc:15]: I assume the `key` is mandatory, right? We should better explain that.

**Review by scholzj — COMMENTED** [review]
Left some nits. Also, we need to maintain backwards compatibility. So the STs should IMHO test both ways - the old and the new.

**Comment by sknot-rh** [comment]
@strimzi-ci run tests profile=regression testcase=LoggingChangeST,RollingUpdateST

**Review by scholzj — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,22 @@
...
+  ```
+  * to
+  ```
+  logging:
+    type: external
+    valueFrom:
+      configMapKeyRef:
+        name: my-config-map
+        key: my-key
+  ```
```
Comment (scholzj) [inline CHANGELOG.md:22]: This should move to the deprecation section.

**Review by scholzj — COMMENTED** [review]
Can you please resolve the conflict so that we can run regression tests?

File: api/src/test/resources/io/strimzi/api/kafka/model/040-Crd-kafka-crdApi-v1.yaml [hunk 5]
```diff
@@ -921,16 +921,28 @@ spec:
...
+                            properties:
+                              key:
+                                type: string
+                              name:
+                                type: string
+                              optional:
+                                type: boolean
+                            description: Reference to the key in the ConfigMap containing
+                              the logging configuration.
+                        description: 'ConfigMap where the Log4j(2) configuration is
```
Comment (PaulRMellor) [inline api/src/test/resources/io/strimzi/api/kafka/model/040-Crd-kafka-crdApi-v1.yaml:944]: We use _log4j2_ in the  guides

File: documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc [hunk 6]
```diff
@@ -12,7 +12,7 @@ The Topic Operator uses the Apache `log4j2` logger implementation.
 Use the `logging` property in the `entityOperator.topicOperator` field of the Kafka resource `Kafka` resource to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc:15]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.EntityTopicOperatorSpec.adoc:15]: Shall we write a line on what default logging means?

File: documentation/api/io.strimzi.api.kafka.model.EntityUserOperatorSpec.adoc [hunk 7]
```diff
@@ -12,7 +12,7 @@ The User Operator uses the Apache `log4j2` logger implementation.
 Use the `logging` property in the `entityOperator.userOperator` field of the `Kafka` resource to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.EntityUserOperatorSpec.adoc:15]: Shall we write a line on what default logging means?

Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.EntityUserOperatorSpec.adoc:15]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

File: api/src/test/resources/io/strimzi/api/kafka/model/040-Crd-kafka-crdApi-v1.yaml (hunk 5 above)
Comment (PaulRMellor) [inline api/src/test/resources/io/strimzi/api/kafka/model/040-Crd-kafka-crdApi-v1.yaml:944]: Check single quotes around descriptions.

Comment (PaulRMellor) [reply api/src/test/resources/io/strimzi/api/kafka/model/040-Crd-kafka-crdApi-v1.yaml:944]: _log4j/2_

File: documentation/api/io.strimzi.api.kafka.model.KafkaBridgeSpec.adoc [hunk 8]
```diff
@@ -58,7 +58,9 @@ The log level of all other operations is set to `INFO` by default.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration.
+The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory.
+Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.KafkaBridgeSpec.adoc:63]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaBridgeSpec.adoc:63]: Shall we write a line on what default logging means?

File: documentation/api/io.strimzi.api.kafka.model.KafkaClusterSpec.adoc [hunk 9]
```diff
@@ -223,7 +223,7 @@ Kafka uses the Apache `log4j` logger implementation.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.KafkaClusterSpec.adoc:226]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the`name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaClusterSpec.adoc]: Shall we write a line on what default logging means?

File: documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc [hunk 10]
```diff
@@ -105,7 +105,7 @@ Kafka Connect uses the Apache `log4j` logger implementation.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc]: Shall we write a line on what default logging means?

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerSpec.adoc [hunk 11]
```diff
@@ -30,7 +30,7 @@ MirrorMaker uses the Apache `log4j` logger implementation.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerSpec.adoc]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerSpec.adoc]: Shall we write a line on what default logging means?

File: documentation/api/io.strimzi.api.kafka.model.ZookeeperClusterSpec.adoc [hunk 12]
```diff
@@ -68,7 +68,7 @@ ZooKeeper uses the Apache `log4j` logger implementation.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.ZookeeperClusterSpec.adoc]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

File: documentation/assemblies/configuring/assembly-external-logging.adoc [hunk 13]
```diff
@@ -25,11 +25,16 @@ spec:
...
+      configMapKeyRef:
+        name: _customConfigMap_
+        key: _keyInConfigMap_
 ----
 
 With external logging, logging properties are defined in a ConfigMap.
-The name of the ConfigMap is referenced in the `spec.logging.name` property.
+The name of the ConfigMap is referenced in the `spec.logging.valueFrom.configMapKeyRef.name` property.
+The `spec.logging.valueFrom.configMapKeyRef.name` and `spec.logging.valueFrom.configMapKeyRef.key` properties are mandatory.
+Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/assemblies/configuring/assembly-external-logging.adoc]:
```suggestion
The name of the ConfigMap is referenced in the `spec.logging.valueFrom.configMapKeyRef.name` property.
Inside the ConfigMap, the logging configuration is described using `log4j.properties` or `log4j2.properties` depending on the component. The `spec.logging.valueFrom.configMapKeyRef.name` and `spec.logging.valueFrom.configMapKeyRef.key` properties are mandatory.
Default logging is used if the `name` or `key` is not set.
```

File: documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc (hunk 10 above)
Comment (sknot-rh) [reply documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc]: What do you suggest?

File: documentation/modules/appendix_crds.adoc [hunk 14]
```diff
@@ -1119,11 +1119,28 @@ The `type` property is a discriminator that distinguishes use of the `ExternalLo
 It must have the value `external` for the type `ExternalLogging`.
 [options="header"]
 |====
-|Property     |Description
-|type  1.2+<.<|Must be `external`.
+|Property          |Description
+|type       1.2+<.<|Must be `external`.
 |string
-|name  1.2+<.<|The name of the `ConfigMap` from which to get the logging configuration.
+|name       1.2+<.<|*The property `name` has been deprecated. This feature should now be configured at path `logging.valueFrom.configMapKeyRef.name`.* The name of the `ConfigMap` from which to get the logging configuration.
```
Comment (PaulRMellor) [inline documentation/modules/appendix_crds.adoc:1125]:
```suggestion
|name       1.2+<.<|*The `name` property has been deprecated, and should now be configured using `logging.valueFrom.configMapKeyRef.name`.* The name is the `ConfigMap` from which to get the logging configuration.
```

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 15]
```diff
@@ -129,7 +129,7 @@ Cruise Control uses the Apache `log4j` logger implementation.
 Use the `logging` property to configure loggers and logger levels.
 
 You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
-If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`.
+If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging will be used if one of `name` or `key` is not set.
```
Comment (PaulRMellor) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc]:
```suggestion
If a ConfigMap is used, you set `logging.valueFrom.configMapKeyRef.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j.properties`. The `logging.valueFrom.configMapKeyRef.name` and `logging.valueFrom.configMapKeyRef.key` properties are mandatory. Default logging is used if the `name` or `key` is not set.
```

Comment (PaulRMellor) [reply documentation/modules/cruise-control/ref-cruise-control-configuration.adoc]: Shall we write a line on what default logging means?

**Review by PaulRMellor — CHANGES_REQUESTED** [review]
Looks good. Few minor suggestions for changes in all instances. As discussed, need to look into the single quotes in the description.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 16]
```diff
@@ -175,6 +192,167 @@ void testJSONFormatLogging() {
...
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(clusterName));
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName));
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(KafkaResources.entityOperatorDeploymentName(clusterName));
+        Map<String, String> operatorSnapshot = DeploymentUtils.depSnapshot(ResourceManager.getCoDeploymentName());
+
+        assertThat(StUtils.checkLogForJSONFormat(operatorSnapshot, ResourceManager.getCoDeploymentName()), is(true));
+        assertThat(StUtils.checkLogForJSONFormat(kafkaPods, "kafka"), is(true));
+        assertThat(StUtils.checkLogForJSONFormat(zkPods, "zookeeper"), is(true));
+        assertThat(StUtils.checkLogForJSONFormat(eoPods, "topic-operator"), is(true));
+        assertThat(StUtils.checkLogForJSONFormat(eoPods, "user-operator"), is(true));
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:240]: How about this?
```suggestion
        assertTrue(StUtils.checkLogForJSONFormat(operatorSnapshot, ResourceManager.getCoDeploymentName()));
        assertTrue(StUtils.checkLogForJSONFormat(kafkaPods, "kafka"));
        assertTrue(StUtils.checkLogForJSONFormat(zkPods, "zookeeper"));
        assertTrue(StUtils.checkLogForJSONFormat(eoPods, "topic-operator"));
        assertTrue(StUtils.checkLogForJSONFormat(eoPods, "user-operator"));
```

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java [hunk 17]
```diff
@@ -571,6 +573,71 @@ void testExternalLoggingChangeTriggerRollingUpdate() {
...
+                "log4j.logger.kafka.network.RequestChannel$=INFO\n" +
+                "log4j.logger.kafka.controller=INFO\n" +
+                "log4j.logger.kafka.log.LogCleaner=INFO\n" +
+                "log4j.logger.state.change.logger=TRACE\n" +
+                "log4j.logger.kafka.authorizer.logger=INFO";
+
+        String configMapLoggersName = "loggers-config-map";
+        ConfigMap configMapLoggers = new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName(configMapLoggersName)
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java:606]: indent

**Review by im-konge — APPROVED** [review]

File: documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc (hunk 10 above)
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc]: Maybe we can change to _If you do not specify a ConfigMap, default ConfigMap settings are used. If a logger is not set, default values are put into the ConfigMap. The ConfigMap is created with the custom resource when the Cluster Operator is started, then recreated after each reconciliation. The name of the ConfigMap takes the form  _CLUSTER-NAME_-_MODEL-NAME_-config, for example, my-cluster-kafka-config._  And then say you can use the name to check the settings.

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaConnectSpec.adoc]: After discussing, let's go with:

_A ConfigMap using the exact logging configuration specified is created with the custom resource when the Cluster Operator is running, then recreated after each reconciliation. If you do not specify a custom ConfigMap, default logging settings are used. If a specific logger value is not set, upper-level logger settings are inherited for that logger._

**Review by PaulRMellor — APPROVED** [review]
What happens when specifying a custom configmap now much clearer :+1:

**Pull request closed without merging**